documentation/OpenAPI/example.rst

```diff
@@ -0,0 +1,187 @@
+                    type:
+                        type: string
+                        description: 'Type of animal the `Fact` describes (e.g. ‘cat’, ‘dog’, ‘horse’)'
+
+This will allow us to request CatFact API and get a random Fact !
+
+.. _Stoplight: https://stoplight.io/studio/
+.. _CatFacts API: https://alexwohlbruck.github.io/cat-facts/
+
+Jane configuration
+------------------
+
+We need to configure Jane before generation. So we create a ``.jane-openapi`` file::
+
+    <?php
+
+    return [
+        'openapi-file' => __DIR__ . '/schema.yaml',
+        'namespace' => 'CatFacts\Api',
+        'directory' => __DIR__ . '/generated/',
+    ];
+
+It will contains a reference to your main schema file (that file can be linked to other files if you want), the PHP
+namespace you want for generated classes and the directory you want to use.
+
+Jane generation
+---------------
+
+Now we can run generation, basically just require jane with composer (see :doc:`/OpenAPI/introduction` for more details
+about installation) and run the following command:
+
+.. code-block:: bash
+
+    vendor/bin/jane-openapi generate
+
+It will find any ``.jane-openapi`` file and use it as configuration. If your file has a different name, just add
+``-c path/to/my/file`` after the command.
+
+Creating a client
+-----------------
+
+Then you need a Client to bridge between Jane and you application. Jane use HTTPlug_ to make this bridge easier
+which allow us to have any middleware we need. Here is a ClientFactory example::
+
+    <?php
+
+    namespace CatFacts;
+
+    use CatFacts\Api\Client;
+    use Http\Client\Common\Plugin\AddHostPlugin;
+    use Http\Client\Common\Plugin\ErrorPlugin;
+    use Http\Client\Common\PluginClient;
+    use Http\Client\HttpClient;
+    use Http\Discovery\HttpClientDiscovery;
+    use Http\Discovery\UriFactoryDiscovery;
+
+    class ClientFactory
+    {
+        public static function create(HttpClient $httpClient = null): Client
+        {
+            // Find a default HTTP client if none provided
+            if (null === $httpClient) {
+                $httpClient = HttpClientDiscovery::find();
+            }
+
+            // Decorates the HTTP client with some plugins
+            $uri = UriFactoryDiscovery::find()->createUri('https://cat-fact.herokuapp.com');
+            $pluginClient = new PluginClient($httpClient, [
+                new ErrorPlugin(),
+                new AddHostPlugin($uri),
+            ]);
+
+            // Instantiate an OpenApi client generated by Jane
+            return Client::create($pluginClient);
+        }
+    }
+
+In this example we can see that we give the HTTPlug the URL of the API and we decorate the client with error & host plugins.
+With both theses, we can have a Client by calling this class ...
+
+.. _HTTPlug: http://httplug.io/
+
+Using your client
+-----------------
+
+Finally we can use our Client and try to get some cool cat fact::
```

Comment:
```suggestion
Finally we can use our Client and try to get some cool cat fact:
```

Reply:
`::` is here to tell RST to parse PHP after it, but will be displayed as `:`